### PR TITLE
feat(component): add separator to dropdown item groups

### PR DESCRIPTION
**Pull request by bc-williamkwon**
Add hr to dropdown when separated flag is true for dropdown groups

image
image

### CODE REVIEW

File: packages/big-design/src/components/Dropdown/spec.tsx [hunk 1]
```diff
@@ -528,3 +528,27 @@ test('renders appropriate amount of list items', async () => {
...
+          items: [{ content: 'Option 2', onItemClick }],
+        },
+      ]}
+      toggle={<Button>Button</Button>}
+    />,
+  );
+  const toggle = getByRole('button');
+  fireEvent.click(toggle);
+
+  const hrItems = await waitForElement(() => container.querySelectorAll('HR'));
```
Comment (chanceaclark) [inline packages/big-design/src/components/Dropdown/spec.tsx:551]:
```suggestion
  const hrItems = await waitForElement(() => container.querySelector('hr'));
```

File: packages/big-design/src/components/Dropdown/Dropdown.tsx [hunk 2]
```diff
@@ -146,6 +147,7 @@ export const Dropdown = memo(
       (group: DropdownItemGroup) => {
         return (
           <>
+            {group.separated && <HR margin="none" />}
```
Comment (chanceaclark) [inline packages/big-design/src/components/Dropdown/Dropdown.tsx:150]: You'll need to throw this into a `aria-hidden` `li` to make it semantic & accessible. Follow a similar pattern to `ListGroupHeader`.

File: packages/docs/PropTables/DropdownPropTable.tsx [hunk 3]
```diff
@@ -192,6 +192,11 @@ const dropdownItemGroupProps: Prop[] = [
     types: 'string',
     description: 'Sets the label for the group.',
   },
+  {
+    name: 'separated',
+    types: 'boolean',
+    description: 'If true adds a line separator above label',
```
Comment (chanceaclark) [inline packages/docs/PropTables/DropdownPropTable.tsx:198]:
```suggestion
    description: 'If true, adds a line separator above the group.',
```

File: packages/big-design/src/components/List/GroupSeparator/ListGroupSeparator.tsx [hunk 4]
```diff
@@ -0,0 +1,11 @@
+import React, { memo } from 'react';
+
+import { HR } from '../../Typography';
+
+export const ListGroupSeparator: React.FC<React.LiHTMLAttributes<HTMLLIElement>> = memo(() => (
+  <li aria-hidden={true}>
+    <HR />
+  </li>
+));
```
Comment (chanceaclark) [inline packages/big-design/src/components/List/GroupSeparator/ListGroupSeparator.tsx:9]: We need to add some of the similar features `ListGroupHeader` has for a11y.

```suggestion
export const ListGroupSeparator: React.FC<React.LiHTMLAttributes<HTMLLIElement>> = memo(() => (
  <li tabIndex={-1} onMouseDown={preventFocus} aria-hidden={true}>
    <HR />
  </li>
));

function preventFocus(event: React.MouseEvent<HTMLLIElement, MouseEvent>) {
  event.preventDefault();
}
```

File: packages/big-design/src/components/Dropdown/spec.tsx [hunk 5]
```diff
@@ -528,3 +528,27 @@ test('renders appropriate amount of list items', async () => {
 
   expect(listItems.length).toBe(3);
 });
+
+test('renders appropriate amount of line separators (HR)', async () => {
```
Comment (chanceaclark) [inline packages/big-design/src/components/Dropdown/spec.tsx:532]: We should also test for correct accessibility roles are enabled for this list item. Essentially test if it has a proper `aria-hidden`, `tabindex`, and can't focus on it.

**Comment by bc-chaz** [comment]
@bc-williamkwon could we include designs (screenshot) on this PR since it's UI?

**Comment by jordan-massingill** [comment]
Would you mind adding some before/after screenshots?

File: packages/big-design/src/components/Dropdown/Dropdown.tsx [hunk 6]
```diff
@@ -146,6 +147,7 @@ export const Dropdown = memo(
       (group: DropdownItemGroup) => {
         return (
           <>
+            {group.separated && <ListGroupSeparator />}
```
Comment (bc-chaz) [inline packages/big-design/src/components/Dropdown/Dropdown.tsx:150]: One issue that we're going to run into using groupings is the space between the groups.  If you look at the designs for storefront manager ui, you'll notice that there isn't really any added space around the line separator.

Storefront UI designs:
Screen Shot 2020-08-27 at 10 45 37 AM

Big Design dropdown w/o group labels:
Screen Shot 2020-08-27 at 10 47 30 AM

Comment (bc-chaz) [reply packages/big-design/src/components/Dropdown/Dropdown.tsx:150]: here's the Figmas for the new UI: https://www.figma.com/file/uFeZH9uVYkPxBcQgCr1ASC/MSF-Merchandising-concept?node-id=1994%3A65969 - Theme Actions -> Step 2 -> Channel Context

Comment (chanceaclark) [reply packages/big-design/src/components/Dropdown/Dropdown.tsx:150]: I believe I fixed that gap in #443.

Comment (bc-chaz) [reply packages/big-design/src/components/Dropdown/Dropdown.tsx:150]: Interesting - when I remove the label in the BD sandbox (https://developer.bigcommerce.com/big-design/dropdown), it still shows the gap. Having some screenshots on this PR would def help

Comment (chanceaclark) [reply packages/big-design/src/components/Dropdown/Dropdown.tsx:150]: Yeah, we haven't release a new version/updated the docs since that change.

**Comment by bc-chaz** [comment]
@bc-williamkwon when you add the additional tests, could you also rebase? looks like we might be out of sync

File: packages/big-design/src/components/Dropdown/spec.tsx [hunk 7]
```diff
@@ -528,3 +528,29 @@ test('renders appropriate amount of list items', async () => {
...
+    />,
+  );
+  const toggle = getByRole('button');
+  fireEvent.click(toggle);
+
+  const hrListItem = await waitForElement(() => container.querySelectorAll('hr')[0].parentElement as HTMLElement);
+  expect(hrListItem.getAttribute('aria-hidden')).toBe('true');
+  expect(hrListItem.getAttribute('tabindex')).toBe('-1');
+  fireEvent.mouseOver(hrListItem);
+  expect(document.activeElement).not.toEqual(hrListItem);
```
Comment (chanceaclark) [inline packages/big-design/src/components/Dropdown/spec.tsx:555]: Let's move this accessibility test into its own test case.

**Pull request merged**